geoips/interfaces/text_based/ascii_palettes.py

```diff
@@ -0,0 +1,181 @@
+    reverse : bool, default=False
+        If True, reverse the colormap
+
+    Returns
+    -------
+    cmap : ListedColormap object
+        If cmap_name not specified, the colormap name will be the os.path.basename
+        of the file.
+
+    Notes
+    -----
+    * Lines preceded by '#' are ignored.
+    * 0-255 or 0-1.0 RGB values (0-255 values are normalized to 0-1.0
+        for matplotlib usage)
+    * One white space delimited RGB value per line
+    """
+    # Read data from ascii file into an NLines by 3 float array, skipping
+    # lines preceded by "#"
+    lines = []
+    with open(fpath) as palette:
+        for line in palette.readlines():
+            if line.strip()[0] != "#":
+                lines += [line]
+
+    if len(lines) == 0:
+        # missing the ascii palette completely, raise an error reporting this.
+        raise AsciiPaletteError(
+            f"Missing Ascii Palette in {fpath}.\n Please define the ascii palette "
+            "before continuing."
+        )
+    carray = numpy.zeros([len(lines), 3])
+    for num, line in enumerate(lines):
+        color_nums = line.strip().split()
+        if len(color_nums) != 3:
+            raise AsciiPaletteError(
+                f"One or more lines of the ascii palette in {fpath} are missing "
+                "one or more values of the 'rgb' triplet. Please fix."
+            )
+        try:
+            carray[num, :] = [float(val) for val in line.strip().split()]
+        except ValueError as e:
+            raise AsciiPaletteError(
+                f"Invalid ascii palette found in {fpath}. See resulting error: {e}."
+            )
+
+    # Normalize from 0-255 to 0.0-1.0
+    if carray.max() > 1.0:
+        carray /= 255.0
+
+    # Test to be sure all color array values are between 0.0 and 1.0
+    if not (carray.min() >= 0.0 and carray.max() <= 1.0):
+        raise AsciiPaletteError("All values in carray must be between 0.0 and 1.0.")
+
+    if reverse is True:
+        carray = numpy.flipud(carray)
+
+    if cmap_name is not None:
+        cmap_name = basename(fpath)
+    cmap = colors.ListedColormap(carray, name=cmap_name)
+    return cmap
+
+
+class AsciiPalettesPlugin(BaseTextPlugin):
+    """Class representing an Ascii Palette Used to Construct a GeoIPS Colormapper."""
+
+    @property
+    def colormap(self):
+        """Ascii Palette derived from 'self.plugin_entry'.
+
+        Returns
+        -------
+        cmap: Matplotlob.colors.ListedColormap
+            - The colormap derived from the ascii palette
+        """
+        if not hasattr(self, "_colormap"):
+            path_to_ascii = str(
+                files(self.plugin_entry["package"]) / self.plugin_entry["relpath"]
+            )
+            self._colormap = from_ascii(path_to_ascii)
+        return self._colormap
+
+    def __call__(self, **kwargs):
+        """Generate a dictionary of colormap information used for linear norm cmaps.
```

Comment:
This will be used for more than just linear norm. We'll also be able to use it for things like log_norm among others. Maybe just "Generate a dictionary of colormap information for use by a colormapper".

Reply:
Made this more generic so different normalization routines can be applied.